extensions/parquet/table/src/main/java/io/deephaven/parquet/table/ParquetTableWriter.java

```diff
@@ -90,40 +92,56 @@ static class GroupingColumnWritingInfo {
      * @param t The table to write
      * @param definition Table definition
      * @param writeInstructions Write instructions for customizations while writing
-     * @param destPathName The destination path
+     * @param destFile The destination to write the parquet file
+     * @param metadataFilePath The destination path to store in the metadata files. This can be different from
+     *        {@code destFile} if we are writing the parquet file to a shadow location first since the metadata should
+     *        always hold the accurate path.
      * @param incomingMeta A map of metadata values to be stores in the file footer
-     * @throws SchemaMappingException Error creating a parquet table schema for the given table (likely due to
-     *         unsupported types)
+     * @param groupingColumnsWritingInfoMap A map of grouping column names to their respective info used for writing
+     *        grouping files
+     * @param metadataFileWriter The writer for the _metadata and _common_metadata files
+     * @param computedCache When we need to perform some computation depending on column data to make a decision
+     *        impacting both schema and written data, we store results in computedCache to avoid having to calculate
+     *        twice. An example is the necessary precision and scale for a BigDecimal column written as a decimal
+     *        logical type.
+     *
      * @throws IOException For file writing related errors
      */
     static void write(
             @NotNull final Table t,
             @NotNull final TableDefinition definition,
             @NotNull final ParquetInstructions writeInstructions,
-            @NotNull final String destPathName,
+            @NotNull final File destFile,
+            @NotNull final File metadataFilePath,
```

Comment:
This feels like a confusing variable name. Since it feels like "the path to the metadata file" rather than "the real path to the destination". `destFilePathForMetadata`. Whatever you decide, please propagate the convention.